packages/calypso-e2e/src/lib/utils/get-test-account-by-feature.ts

```diff
@@ -0,0 +1,72 @@
+type Env = 'edge' | 'stable';
+
+type SiteType = 'simple' | 'atomic';
+
+type Feature = 'gutenberg' | 'coblocks';
+type FeatureKey = { [ key in Feature ]?: Env | undefined } & { siteType: SiteType };
+export type FeatureCriteria = FeatureKey & { accountName: string };
+type FeatureMap = Map< string, string >;
+
+function stringifyKey( o: FeatureKey ) {
+	const keys = Object.keys( o ) as [ keyof FeatureKey ];
+	const sorted = keys.sort().reduce( ( sorted, key ) => {
+		//sorted[ key ] = o[ key ];
+		sorted[ key ] = o[ key ];
+		return sorted;
+	}, {} as any ) as FeatureKey;
+
+	return JSON.stringify( sorted );
+}
+
+function criteriaToMap( criteria: FeatureCriteria[], map: FeatureMap ): FeatureMap {
+	return criteria.reduce( ( acc, criteria ) => {
+		const { accountName, ...rest } = criteria;
+		acc.set( stringifyKey( rest ), accountName );
+		return acc;
+	}, map );
+}
+
+// Rename to accountsTable or something along these lines?
+const defaultCriteria: FeatureCriteria[] = [
```

Comment:
I'd like to clarify the relationship between this `defaultCriteria` and the one that might be passed as the second argument for the `getTestAccountByFeature`. For example, if you look at this test: https://github.com/Automattic/wp-calypso/pull/61783/files#diff-62d89bc54131b1f1bba8fd14ded781bc4a3fd8336a0a479fe1d311eda0b5b994, you'll notice that while it uses the `gutenbergSimpleSiteEdgeUser` as the edge account (which is what we consider the "default" one here), for stable it uses `simpleSitePersonalPlanUser` instead of `gutenbergSimpleSiteEdgeUser`. There are probably good reasons for that, and we should support it. To support that, you'll see that in the aforementioned test I'm passing a custom criterion that replaces the default defined here.

Perhaps having a default that's used implicitly could cause more harm than good? Perhaps it'd be better to always pass a `criteria` to the `getTestAccountByFeature` function inline where it's used in the test? Maybe we could still export this const...

Another possibility is to centralize all the possible criteria here, but for that, we'd probably need an additional attribute, like variation. So for example, we could have:

```typescript
const defaultCriteria: FeatureCriteria[] = [
	{
		gutenberg: 'edge',
		siteType: 'simple',
		accountName: 'gutenbergSimpleSiteEdgeUser',
	},
    {
		gutenberg: 'edge',
		siteType: 'simple',
		accountName: 'gutenbergSimpleSiteUser',
	},
	{
		gutenberg: 'edge',
		siteType: 'simple',
        variation: 'siteEditor',
		accountName: 'siteEditorSimpleSiteUser',
	},
    {
		gutenberg: 'edge',
		siteType: 'simple',
        variation: 'siteEditor',
		accountName: 'siteEditorSimpleSiteEdgeUser',
	},
];
```

We could then use it like this:

```js
const accountName = getTestAccountByFeature({gutenberg: 'edge', siteType: 'simple'});
// returns "siteEditorSimpleSiteUser"
const accountName = getTestAccountByFeature({gutenberg: 'edge', siteType: 'simple', variation: 'siteEditor'});
// returns "siteEditorSimpleSiteEdgeUser'"
```

We could still allow the function to accept a second `criteria` argument to augment the internal map with more options - inline - if needed:

```js
const accountName = getTestAccountByFeature({gutenberg: 'edge', siteType: 'simple', variation: 'newOne'}, [{gutenberg: 'edge', siteType: 'simple', variation: 'newOne', accountName: 'kawabanga'}]);
// returns "kawabanga"
);
```

What do you think?

Reply:
I implemented support for this. There's now an additional (and optional) `variation` attribute as part of the `FeatureKey` type. We can use that to keep all `feature -> site` definitions (aka criteria) in a central place (probably [here](https://github.com/Automattic/wp-calypso/pull/61783)?).

Reply:
Would it make things simpler if we drop the use of `simpleSitePersonalPlanUser`?

The `simpleSiteXYZ` users were introduced some months ago as a way to distribute the testing load to multiple users. However, with discussion of WordPress Plans reorganization and use of preauthenticated cookies we can probably go back to using a smaller number of accounts.

Reply:
Sure, having fewer sites is better, but not strictly necessary in this case. I was mainly thinking out loud and documenting how it worked, and I already added support for the additional attribute, so we can have a central list of sites if we want (and avoid [this](https://github.com/Automattic/wp-calypso/pull/61783/files#diff-4e04b7a65aac2ec3d88e111ac4b84cdc060dde17cd7987876a28280e23313a9dR36-R37)). The goal of this is to provide a clean and simple way of selecting an account from a variety of possible combinations based on external data passed to the test.